serve/mlc_serve/engine/staging_engine_worker.py

```diff
@@ -318,6 +324,25 @@ def _adjust_batch(self):
                 self.cache_manager.allocate(state.request_id, num_tokens)
                 self.current_batch[state.request_id] = state
 
+            if not self.current_batch:
```

Comment:
Do we understand why this would occur?  My concern is that the container could get into a state where continuously reject all requests.

Reply:
I'd say we don't. If we knew, we would eliminate such possibility. We tried our best in https://github.com/octoml/mlc-llm/pull/89 but we can never be sure that a hang would never occur.